packages/icon/src/vaadin-icon.js

```diff
@@ -245,6 +273,21 @@ class Icon extends ThemableMixin(ElementMixin(ControllerMixin(ConditionalResizeM
     return viewBox || `0 0 ${size} ${size}`;
   }
 
+  /** @private */
+  __fontChanged(font) {
+    this.classList.remove(...(this.__addedClasses || []));
+    if (font) {
+      this.__addedClasses = font.split(' ');
+      this.classList.add(...this.__addedClasses);
+    }
+
+    // Need to have the "icon" attribute set on the host also when using font icons
+    // to avoid issues such as https://github.com/vaadin/web-components/issues/6301
+    if (font && !this.icon) {
+      this.icon = '';
+    }
```

Comment:
Should it set `this.icon = null` when `font` is falsy?

Reply:
I was thinking about it but couldn't find a reason to do it. Also, in an unlikely scenario where a previously truthy `font` gets set falsy, we can't know if the value of `icon` is set by something else (user, src property, font-family property...).

I'd actually be in favor of making the default value of `icon` an empty string so there would always be an "icon" attribute with `<vaadin-icon>`. Can't see much use for `<vaadin-icon>` without an icon.

Reply:
Yeah... makes sense. Maybe we can have the "icon" attribute by default.

Reply:
Changing the default value of a property could be considered breaking by some so I'd prefer discussing it separately from this PR.

Reply:
Yeah. I understand it. I didn't mean to change it in this PR, though.